server/admin_management/api/app/routers/router_guards.py

```diff
@@ -1,17 +1,41 @@
 import logging
 from http import HTTPStatus
-from fastapi import Depends, HTTPException
+from fastapi import Depends, HTTPException, Request
+from sqlalchemy.orm import Session
+from typing import Union
+import json
 
+from api.app import database
 from api.app.jwt_validation import (
     ERROR_PERMISSION_REQUIRED,
     get_access_roles,
+    get_request_cognito_user_id,
     validate_token,
 )
-
+from api.app.schemas import Requester, TargetUser
+from api.app.models.model import FamUser
+from api.app.services.application_admin_service import ApplicationAdminService
+from api.app.services.user_service import UserService
 
 LOGGER = logging.getLogger(__name__)
 
 
+ERROR_SELF_GRANT_PROHIBITED = "self_grant_prohibited"
+ERROR_INVALID_APPLICATION_ID = "invalid_application_id"
+ERROR_INVALID_ROLE_ID = "invalid_role_id"
+ERROR_REQUESTER_NOT_EXISTS = "requester_not_exists"
+ERROR_EXTERNAL_USER_ACTION_PROHIBITED = "external_user_action_prohibited"
+
+
+no_requester_exception = HTTPException(
+    status_code=HTTPStatus.FORBIDDEN,  # 403
+    detail={
+        "code": ERROR_REQUESTER_NOT_EXISTS,
+        "description": "Requester does not exist, action is not allowed",
+    },
+)
+
+
 def authorize_by_fam_admin(claims: dict = Depends(validate_token)):
     required_role = "FAM_ACCESS_ADMIN"
```

Comment:
Use "FAM_ACCESS_ADMIN" for now, later on when work on the cleanup ticket, we can change it to "FAM_ADMIN". Just need to use "FAM_ACCESS_ADMIN" for now so we can test our endpoints.

Another question is that I rethink about Ian's comment, the required role should be "FAM_ADMIN" or "{application_name}_ADMIN"? so do we allow a "FOM_DEV_ADMIN" user to add another user be FOM dev admin?

Reply:
Got it. Then we will refactor that time.